grin/test/ExtendedSyntax/PrimOpsSpec.hs

```diff
@@ -0,0 +1,241 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE QuasiQuotes #-}
+module ExtendedSyntax.PrimOpsSpec where
+
+import Test.Hspec
+
+import qualified Language.C.Inline as C
+import qualified Language.C.Inline.Unsafe as CU
+import Foreign.C.String
+import Foreign.Marshal.Alloc
+import Test.QuickCheck.Arbitrary (arbitrary)
+import Test.QuickCheck.Gen
+import Test.QuickCheck.Monadic
+
+
+C.include "<stdio.h>"
+-- TODO: Once, we merge the syntax tranformation PR,
+-- remove the extra ".." from here. It is needed now
+-- because of the nested directory structure.
+-- TODO: Once merged, use "../../prim_ops.c"
+C.include "../../prim_ops.h"
+
+spec :: Spec
```

Comment:
With a little more effort you could write property based tests for these primitive functions as well.

Reply:
There are some, see below:
```
    it "works for a random string" $ monadicIO $ do
       str <- pick randomString
       run $ primStringLen str
```

Reply:
Well at least we check if we don't do segfault :D .

Reply:
The next thing would be looking into how a set of primops and their composition do satisfy some kind of commutative diagrams, but that is a bit overkill atm.

Reply:
Good point about the property-based tests. For now, let's just merge this, and address it separately.